packages/main/src/plugin/authentication.ts

```
@@ -0,0 +1,189 @@
+
+  addAccountUsage(providerId: string, accountLabel: string, extensionId: string, extensionName: string): void {
+    throw new Error('The method is not implemented!');
+  }
+
+  readAllowedExtensions(providerId: string, accountName: string): AllowedExtension[] {
+    throw new Error('The method is not implemented!');
+  }
+
+  updateAllowedExtension(
+    providerId: string,
+    accountName: string,
+    extensionId: string,
+    extensionName: string,
+    isAllowed: boolean,
+  ): void {
+    throw new Error('The method is not implemented!');
+  }
+
+  isAccessAllowed(providerId: string, accountName: string, extensionId: string): boolean | undefined {
+    return true; // To be implemented later
+  }
+
+  async getSession(
+    requestingExtension: ExtensionInfo,
+    providerId: string,
+    scopes: string[],
+    options: AuthenticationGetSessionOptions & { createIfNone: true },
+  ): Promise<AuthenticationSession | undefined>;
+  async getSession(
+    requestingExtension: ExtensionInfo,
+    providerId: string,
+    scopes: string[],
+    options?: AuthenticationGetSessionOptions,
+  ): Promise<AuthenticationSession | undefined>;
+  async getSession(
+    requestingExtension: ExtensionInfo,
+    providerId: string,
+    scopes: string[],
+    options: AuthenticationGetSessionOptions = {},
+  ): Promise<AuthenticationSession | undefined> {
+    // Error cases
+    if (options.forceNewSession) {
+      throw new Error('Option is not supported. Please remove forceNewSession option.');
+    }
+    if (options.silent) {
+      throw new Error('Option is not supported. Please remove silent option.');
+    }
+    if (options.clearSessionPreference) {
+      throw new Error('Option is not supported. Please remove clearSessionPreference option.');
+    }
+
+    const provider = this._authenticationProviders.get(providerId)?.provider;
+    if (!provider) {
+      // window.showInformationMessage(`Requested authentication provider ${providerId} is not installed.`);
```

shouldn't throw an error ?
```
Rejects if a provider with providerId is not registered, or if the user does not consent to sharing authentication information with
```

---

It should, but I realized this was the reason for tests errors I mentioned before.

---

ok so let's throw the error to have the correct expectation and then we fix the test if there is a test problem. Shouldn't be a big deal I can also help on the test failure

---

I can switch to Dialogs, so there should be no issue I had.

---

AFAIK it should send an error here, not display a modal
Also we are not inside an extension or on frontend so no 'window' object

It's the client that can display something